DragAndDrop/src/main/java/com/microsoft/device/display/samples/draganddrop/fragment/DropTargetFragment.kt

```diff
@@ -6,206 +6,210 @@
-                setElevation(HINT_ELEVATION, binding.imageHintImageView, binding.imageHintTextView)
-            }
+    private fun setImageTarget(bitmap: Bitmap) {
+        viewModel.image.value = bitmap
+        binding.emptyImage.setImageBitmap(bitmap)
+        binding.emptyImage.background = null
+        binding.dropImageHint.isVisible = false
+    }
 
-            MimeType.TEXT_PLAIN -> {
-                binding.dropTextHint.updateColorAndElevation(colorFilter, HINT_ELEVATION)
-                setElevation(HINT_ELEVATION, binding.textHintImageView, binding.textHintTextView)
-            }
+    private fun handlePlainTextDrop(item: ClipData.Item) {
+        val text = if (item.text != null) {
+            // The text is contained in the ClipData.Item
+            item.text
+        } else {
+            // The text is in a file pointed to by the ClipData.Item
+            getTextFromClipData(item)
         }
+        setTextTarget(text)
     }
 
-    /**
-     * Clears the background color for the drop zone depending on the given mime type
-     */
-    private fun clearBackgroundColor(mimeType: MimeType) {
-        when (mimeType) {
-            MimeType.IMAGE_JPEG -> {
-                binding.dropImageHint.updateColorAndElevation(null, DEFAULT_ELEVATION)
-                setElevation(
-                    DEFAULT_ELEVATION,
-                    binding.imageHintImageView,
-                    binding.imageHintTextView
-                )
-            }
-
-            MimeType.TEXT_PLAIN -> {
-                binding.textHintImageView
-                binding.dropTextHint.updateColorAndElevation(null, DEFAULT_ELEVATION)
-                setElevation(DEFAULT_ELEVATION, binding.textHintImageView, binding.textHintTextView)
-            }
+    private fun handleImageDrop(item: ClipData.Item) {
+        item.uri?.let { uri ->
+            val bitmap =
+                BitmapFactory.decodeStream(requireContext().contentResolver.openInputStream(uri))
+            setImageTarget(bitmap)
+        } ?: run {
+            Log.e(TAG, "Clip data is missing it's URI")
         }
     }
 
-    /**
-     * Adds the text from the [DragEvent.getClipData] to the drop zone
-     *
-     * @param event The [DragEvent] containing the text that will be added to the drop zone
-     */
-    private fun handleTextDrop(event: DragEvent) {
-        val item = event.clipData.getItemAt(0)
-        val dragData = item.text.toString()
-        var sourceView = event.localState as? View
-        sourceView?.let {
-            it.remove()
-        } ?: run {
-            sourceView = TextView(requireContext()).apply {
-                text = dragData
-            }
+    private fun getTextFromClipData(item: ClipData.Item): String {
+        val parcelFileDescriptor: ParcelFileDescriptor? = try {
+            requireActivity().contentResolver.openFileDescriptor(item.uri, "r")
+        } catch (e: FileNotFoundException) {
+            e.printStackTrace()
```

Comment:
Maybe you should use Log.e also here, like you already did at 183 line?

Reply:
Yes, please. We never should use `exception.printStackTrace` in our code. Use always `Log.xxx`.